src/Evolution/DgSubcell/NeighborReconstructedFaceSolution.hpp

```diff
@@ -0,0 +1,164 @@
+            std::pair<Direction<Metavariables::volume_dim>,
+                      ElementId<Metavariables::volume_dim>>,
+            std::tuple<Mesh<Metavariables::volume_dim - 1>,
+                       std::optional<std::vector<double>>,
+                       std::optional<std::vector<double>>, ::TimeStepId>,
+            boost::hash<std::pair<Direction<Metavariables::volume_dim>,
+                                  ElementId<Metavariables::volume_dim>>>>>*>
+        received_temporal_id_and_data) noexcept {
+  constexpr size_t volume_dim = Metavariables::volume_dim;
+  db::mutate<
+      subcell::Tags::NeighborDataForReconstructionAndRdmpTci<volume_dim>>(
+      box, [&received_temporal_id_and_data](
+               const auto subcell_neighbor_data_ptr) noexcept {
+        subcell::NeighborData& self_neighbor_data =
+            subcell_neighbor_data_ptr->at(
+                std::pair{Direction<volume_dim>::lower_xi(),
+                          ElementId<volume_dim>::external_boundary_id()});
+        const size_t number_of_evolved_vars =
+            self_neighbor_data.max_variables_values.size();
+        for (auto& received_mortar_data :
+             received_temporal_id_and_data->second) {
+          const auto& mortar_id = received_mortar_data.first;
+          ASSERT(std::get<1>(received_mortar_data.second).has_value(),
+                 "The subcell mortar data was not sent at TimeStepId "
+                     << received_temporal_id_and_data->first
+                     << " with mortar id (" << mortar_id.first << ','
+                     << mortar_id.second << ")");
+          const std::vector<double>& neighbor_ghost_and_subcell_data =
+              *std::get<1>(received_mortar_data.second);
+          // Compute min and max over neighbors
+          const size_t offset_for_min =
+              neighbor_ghost_and_subcell_data.size() - number_of_evolved_vars;
+          const size_t offset_for_max = offset_for_min - number_of_evolved_vars;
+          for (size_t var_index = 0; var_index < number_of_evolved_vars;
+               ++var_index) {
+            self_neighbor_data.max_variables_values[var_index] = std::max(
+                self_neighbor_data.max_variables_values[var_index],
+                neighbor_ghost_and_subcell_data[offset_for_max + var_index]);
+            self_neighbor_data.min_variables_values[var_index] = std::min(
+                self_neighbor_data.min_variables_values[var_index],
+                neighbor_ghost_and_subcell_data[offset_for_min + var_index]);
+          }
+
+          ASSERT(subcell_neighbor_data_ptr->find(mortar_id) ==
+                     subcell_neighbor_data_ptr->end(),
+                 "The subcell neighbor data is already inserted. Direction: "
+                     << mortar_id.first
+                     << " with ElementId: " << mortar_id.second);
+          // Copy over the neighbor data for reconstruction. We need this
+          // since we might be doing a step unwind.
```

Comment:
Can you expand on this a little? does `neighbor_ghost_and_subcell_data` get erased when unwinding? I'm not quite sure what is meant here so a few more words would help I think.